patroni/dcs/__init__.py

```diff
@@ -954,10 +998,11 @@ def attempt_to_acquire_leader(self):
         process requests (hopefuly temporary), the ~DCSError exception should be raised"""
 
     @abc.abstractmethod
-    def set_failover_value(self, value, index=None):
+    def set_failover_value(self, value: str, index: Optional[Any] = None) -> bool:
         """Create or update `/failover` key"""
 
-    def manual_failover(self, leader, candidate, scheduled_at=None, index=None):
+    def manual_failover(self, leader: str, candidate: str, scheduled_at: Optional[datetime.datetime] = None,
```

Comment:
```suggestion
    def manual_failover(self, leader: Optional[str], candidate: Optional[str], scheduled_at: Optional[datetime.datetime] = None,
```

The `if leader` and `if candidate` that are performed in the function body give me the impression the parameters can assume `None` value.